### PR TITLE
[Rule Tuning] Add Initial SentinelOne Compatibility to Windows DRs

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Resolves https://github.com/elastic/ia-trade-team/issues/243

## Summary

Adds compatibility to S1 cloud funnel logs to the following rules:

- Exporting Exchange Mailbox via PowerShell
- Port Forwarding Rule Addition
- Potential Remote Desktop Tunneling Detected
- Suspicious ScreenConnect Client Child Process
- Creation or Modification of Domain Backup DPAPI private key
- Kirbi File Creation
- Mimikatz Memssp Log File Detected
- Suspicious Antimalware Scan Interface DLL
- Encoded Executable Stored in the Registry
- Suspicious WerFault Child Process
- Local Account TokenFilter Policy Disabled
- Suspicious Zoom Child Process
- Unusual Parent Process for cmd.exe
- Suspicious MS Outlook Child Process
- Adobe Hijack Persistence
- Registry Persistence via AppCert DLL
- Persistence via Scheduled Job Creation
- System Shells via Services
- Persistence via BITS Job Notify Cmdline
- Web Shell Detection: Script Process Child of Common Web Processes

Screenshots:

<details> 

- Exporting Exchange Mailbox via PowerShell
Screenshot: 
image

- Port Forwarding Rule Addition
Screenshot: 
image

- Potential Remote Desktop Tunneling Detected
Screenshot: 
image

- Suspicious ScreenConnect Client Child Process
Screenshot: 
image

- Creation or Modification of Domain Backup DPAPI private key
Screenshot: 
image

- Kirbi File Creation
Screenshot: 
image

- Mimikatz Memssp Log File Detected
Screenshot: 
image

- Local Account TokenFilter Policy Disabled
Screenshot: 
image

- Suspicious Zoom Child Process
Screenshot: 
image

- Unusual Parent Process for cmd.exe
Screenshot: 
image

- Suspicious MS Outlook Child Process
Screenshot: 
image

- Adobe Hijack Persistence
Screenshot: 
image

- Registry Persistence via AppCert DLL
Screenshot: 
image

- System Shells via Services
Screenshot: 
image

- Persistence via BITS Job Notify Cmdline
Screenshot: 
image

- Web Shell Detection: Script Process Child of Common Web Processes
Screenshot: 
image

</details>

The following are being added without an event in the stack, based in the fields used:

- Suspicious Antimalware Scan Interface DLL
- Encoded Executable Stored in the Registry
- Suspicious WerFault Child Process
- Persistence via Scheduled Job Creation

### CODE REVIEW

**Review by Aegrah — APPROVED** [review]
Other than the unit test integration list issue, LGTM

**Review by terrancedejesus — APPROVED** [review]
@w0rk3r fantastic work! Thank you for taking the time to include all the screenshots as well.

@Mikaayenson @eric-forte-elastic @brokensound77 - No CTA now, but I have a feeling we may want to explore some resiliency to potentially breaking rules that are compatible with multiple 3rd-party integrations. If we need to make a change, how can we be sure we will not break the rule across every EDR and does everyone have access to that. @w0rk3r may be something we can follow-up on in the Epic if that's alright.

**Comment by terrancedejesus** [comment]
@w0rk3r do we need to pull the integration and manifests to do proper schema validation here? It will reference the integration tag and want to ensure those fields exist in the S1 integration. We should also confirm that the right S1 integration version and available stack match the min-stack defined in these rules.

**Review by terrancedejesus — APPROVED** [review]
Looking good! Great work getting these compatible 🚀

**Comment by terrancedejesus** [comment]
@brokensound77 @eric-forte-elastic @Mikaayenson - We need a code owner to review and approve. Please advise when you have time. Thank you.

**Review by Mikaayenson — APPROVED** [review]
LGTM.!

**Comment by terrancedejesus** [comment]
@Mikaayenson @eric-forte-elastic - Although approved, please review the code changes I made in `rule.py` and `integrations.py`. These changes were necessary as the schema updates happened in a pre-release version of the Sentinel One Cloud Funnel integration, whose minimum Kibana version was set to 8.10.1 in the manifest. When it went GA, the schema mappings were also adjusted to an ECS mappings feature that is only available in Kibana version 8.13.0. Thus the only GA version is 1.0.0 at 8.13, which means these rule updates would have to min-stack at 8.13 or no compatible integration manifest and schema could be identified.

Since we exposed the option to add the pre-releases in the CLI, I re-ran manifest and schema generation so that our Sentinel One Cloud Funnel integration manifests and schemas have both GA and pre-release in them, thus the rules validate properly and can backport to at least 8.10.

File: detection_rules/schemas/definitions.py [hunk 1]
```diff
@@ -33,7 +33,14 @@
...
 }
-NON_DATASET_PACKAGES = ['apm', 'auditd_manager', 'cloud_defend', 'endpoint', 'network_traffic', 'system', 'windows']
+NON_DATASET_PACKAGES = ['apm',
+                        'auditd_manager',
+                        'cloud_defend',
+                        'endpoint',
+                        'network_traffic',
+                        'system',
+                        'windows',
+                        'sentinel_one_cloud_funnel']
```
Comment (Mikaayenson) [inline detection_rules/schemas/definitions.py:43]: @w0rk3r Is there any reason why we didn't use the [sentinel_one](https://github.com/elastic/integrations/tree/main/packages/sentinel_one) integration instead?

File: rules/windows/collection_email_powershell_exchange_mailbox.toml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 [metadata]
 creation_date = "2020/12/15"
-integration = ["endpoint", "windows", "system"]
+integration = ["endpoint", "windows", "system", "sentinel_one_cloud_funnel"]
 maturity = "production"
 min_stack_comments = "New fields added: required_fields, related_integrations, setup"
-min_stack_version = "8.3.0"
-updated_date = "2024/03/28"
+min_stack_version = "8.10.1"
```
Comment (Mikaayenson) [inline rules/windows/collection_email_powershell_exchange_mailbox.toml:6]: Do we have to align this to the stack-schema-map?

**Review by terrancedejesus — CHANGES_REQUESTED** [review]
min-stack at 8.11

File: detection_rules/schemas/definitions.py (hunk 1 above)
Comment (w0rk3r) [reply detection_rules/schemas/definitions.py:43]: It doesn't collect EDR telemetry, it focuses in the alerts and related data.

**Review by terrancedejesus — APPROVED** [review]
Thanks Jon!

**Comment by Mikaayenson** [comment]
Just adding for posterity. We had to min stack at 8.11 because the integration did something unique when going to GA which required stack version to be ^8.13. The only available prerelease version available was at 8.10.1 but we dont support patch versions in our validation so the next version available was 8.11.

**Pull request merged**